Submitted By: William Harrington <kb0iic@hotmail.com>
Date: 2005-08-24
Initial Package Version: 1.4.2
Upstream Status: Submitted
Origin: William Harrington 
Description: Fixed order of processing so declaration is after structure.
	     Fixed invalid lvalue in assignment.

diff -Naur inetutils-1.4.2.orig/ftp/extern.h inetutils-1.4.2/ftp/extern.h
--- inetutils-1.4.2.orig/ftp/extern.h	2000-08-09 00:44:23.000000000 -0500
+++ inetutils-1.4.2/ftp/extern.h	2005-08-24 19:43:56.000000000 -0500
@@ -137,7 +137,6 @@
 
 extern jmp_buf	abortprox;
 extern int	abrtflag;
-extern struct	cmd cmdtab[];
 extern FILE	*cout;
 extern int	data;
 extern char    *home;
diff -Naur inetutils-1.4.2.orig/ftp/ftp_var.h inetutils-1.4.2/ftp/ftp_var.h
--- inetutils-1.4.2.orig/ftp/ftp_var.h	2000-07-07 20:00:53.000000000 -0500
+++ inetutils-1.4.2/ftp/ftp_var.h	2005-08-24 19:44:15.000000000 -0500
@@ -120,6 +120,8 @@
 	void	(*c_handler) __P((int, char **)); /* function to call */
 };
 
+extern struct   cmd cmdtab[];
+
 struct macel {
 	char mac_name[9];	/* macro name */
 	char *mac_start;	/* start of macro in macbuf */
diff -Naur inetutils-1.4.2.orig/libinetutils/ttymsg.c inetutils-1.4.2/libinetutils/ttymsg.c
--- inetutils-1.4.2.orig/libinetutils/ttymsg.c	2001-11-01 09:52:19.000000000 -0600
+++ inetutils-1.4.2/libinetutils/ttymsg.c	2005-08-24 19:43:35.000000000 -0500
@@ -132,7 +132,7 @@
 	    }
 	  if (wret)
 	    {
-	      (char *)iov->iov_base += wret;
+	      iov->iov_base = (char *)iov->iov_base + wret;
 	      iov->iov_len -= wret;
 	    }
 	  continue;
